x/btcstaking/keeper/msg_server_test.go

```diff
@@ -159,13 +159,20 @@ func FuzzCreateBTCDelegation(f *testing.F) {
 
 		// generate and insert new BTC delegation
 		stakingValue := int64(2 * 10e8)
-		stakingTxHash, _, _, msgCreateBTCDel, _ := h.CreateDelegation(
+		delSK, _, err := datagen.GenRandomBTCKeyPair(r)
+		h.NoError(err)
+		stakingTxHash, msgCreateBTCDel, _, _, _, err := h.CreateDelegation(
 			r,
+			delSK,
 			fpPK,
 			changeAddress.EncodeAddress(),
 			stakingValue,
 			1000,
+			0,
+			0,
+			false,
```

Comment:
wdyt about also having this randomized ? Sometimes we will hit pre-approval flow, sometimes old flow, eitherway we need to test both codepaths. 

This applies also to other tests like `FuzzAddCovenantSigs`

Also it would be great to fuzz test for the new message for providing inclusion proof.

Reply:
Sounds good. I have randomised `FuzzCreateBTCDelegation` and `FuzzAddCovenantSigs` which should cover the lifecycle of pending -> verified -> active (no pre-approval), and added a new fuzz test `FuzzAddBTCDelegationInclusionProof` for the inclusion proof. The fuzz tests in voting power table should have covered the whole lifecycle of pre-approval as well.